codice/ddf
# DDF-5431 Text Product Preview With Metadata

#### What does this PR do?
For users who are working with textual products rather than imagery. Report text, message traffic, news feeds, etc. are often embedded directly in the metadata. We want to expose that information in the preview tab.

#### Who is reviewing it? 
@brendan-hofmann 
@aaronilovici
@mdang8 
@rfding 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@oconnormi

#### How should this be tested?
1. Pull and build branch
2. Go to the Admin Console and run the setup
3. Go the the search UI, ingest a text-based file, then search for it in a workspace. Verify that the Inspector *does not* have a preview tab.
4. Go to the **Catalog** app in the Admin Console
5. Select the Configuration tab, then select **Catalog Preview** from the list and enable "Preview From Metadata"
6. Go back to the search UI and search for the ingested file again. The Inspector should have a Preview tab with the files text-content in it.


#### What are the relevant tickets?
Fixes: #__5431__

#### Screenshots

Metatypes
[
<img width="693" alt="Screen Shot 2019-10-16 at 12 28 09 PM" src="https://user-images.githubusercontent.com/11358088/66939206-9f0bbb80-f010-11e9-95e2-f6c0b9d02b7a.png">
](url)
Preview Tab
[
<img width="1459" alt="Screen Shot 2019-10-16 at 12 28 32 PM" src="https://user-images.githubusercontent.com/11358088/66939225-a7fc8d00-f010-11e9-8c60-84f403597bbc.png">
](url)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
